## Compare CORE-main labels with GINCO

In [1]:
import pandas as pd
import numpy as np

In [4]:
# Import the GINCO dataset
ginco = pd.read_csv("data-sheets-with-all-info/GINCO-MT-GINCO-keeptext-with-all-information.csv", sep="\t", index_col=0)
ginco.head(3)

id                                                url  crawled   hard  \
0  3949  http://www.pomurje.si/aktualno/sport/zimska-li...     2014  False   
1  3726  http://www.ss-sezana.si/sss/index.php?option=c...     2014  False   
2  5621  http://www.kamnik-starejsi.si/novice/144-sodel...     2014  False   

           primary_level_1          primary_level_2          primary_level_3  \
0           News/Reporting           News/Reporting           News/Reporting   
1  Information/Explanation  Information/Explanation  Information/Explanation   
2    Promotion of Services    Promotion of Services                Promotion   

       secondary_level_1      secondary_level_2      secondary_level_3  ...  \
0                    NaN                    NaN                    NaN  ...   
1                    NaN                    NaN                    NaN  ...   
2  Opinion/Argumentation  Opinion/Argumentation  Opinion/Argumentation  ...   

                                        Slovene_text  \
0  Šport <p/> Zimska liga malega nogometa sobota,...   
1  JEDILNIK <p/> Iskalnik <p/> Poglavitni cilj pr...   
2  Projekt INNOVAge in zavod Oreli <p/> Zavod Ore...   

                                             MT_text text_length  \
0  Sport <p/> Winter Little League Football Satur...          93   
1  JEDILNIK <p/> Search <p/> The main objective o...          76   
2  Project INNOVAge and the Oreli Institute <p/> ...         232   

      FTD_pred_on_SL     FTD_pred_on_MT split-without-rare-categories  \
0          A8 (news)          A8 (news)                          test   
1  A16 (information)  A16 (information)                          test   
2    A12 (promotion)    A12 (promotion)                         train   

           primary_level_4 downcast_split  \
0           News/Reporting           test   
1  Information/Explanation            dev   
2                Promotion           test   

                    CORE_main_pred_on_SL  \
0                              Narrative   
1  Informational Description/Explanation   
2  Informational Description/Explanation   

                    CORE_main_pred_on_MT  
0                              Narrative  
1  Informational Description/Explanation  
2  Informational Description/Explanation  

[3 rows x 26 columns]

In [7]:
# Analyse how GINCO primary labels are connected with CORE labels based on the prediction
pd.crosstab(ginco["primary_level_1"], ginco["CORE_main_pred_on_MT"], normalize="index")


CORE_main_pred_on_MT        How-To/Instructional  \
primary_level_1                                    
Announcement                            0.000000   
Call                                    0.000000   
Correspondence                          0.000000   
FAQ                                     0.000000   
Forum                                   0.000000   
Information/Explanation                 0.000000   
Instruction                             0.500000   
Interview                               0.000000   
Invitation                              0.000000   
Legal/Regulation                        0.000000   
List of Summaries/Excerpts              0.009434   
Lyrical                                 0.000000   
News/Reporting                          0.000000   
Opinion/Argumentation                   0.017544   
Opinionated News                        0.000000   
Other                                   0.000000   
Promotion                               0.033333   
Promotion of Services                   0.000000   
Promotion of a Product                  0.000000   
Prose                                   0.000000   
Recipe                                  1.000000   
Research Article                        0.000000   
Review                                  0.000000   
Script/Drama                            0.000000   

CORE_main_pred_on_MT        Informational Description/Explanation  \
primary_level_1                                                     
Announcement                                             0.941176   
Call                                                     1.000000   
Correspondence                                           0.312500   
FAQ                                                      0.666667   
Forum                                                    0.076923   
Information/Explanation                                  0.907692   
Instruction                                              0.447368   
Interview                                                0.000000   
Invitation                                               0.968750   
Legal/Regulation                                         1.000000   
List of Summaries/Excerpts                               0.424528   
Lyrical                                                  0.000000   
News/Reporting                                           0.373913   
Opinion/Argumentation                                    0.333333   
Opinionated News                                         0.247191   
Other                                                    0.470588   
Promotion                                                0.700000   
Promotion of Services                                    0.875000   
Promotion of a Product                                   0.513043   
Prose                                                    0.000000   
Recipe                                                   0.000000   
Research Article                                         1.000000   
Review                                                   0.176471   
Script/Drama                                             1.000000   

CORE_main_pred_on_MT        Informational Persuasion  Interactive Discussion  \
primary_level_1                                                                
Announcement                                0.000000                0.000000   
Call                                        0.000000                0.000000   
Correspondence                              0.062500                0.312500   
FAQ                                         0.000000                0.000000   
Forum                                       0.000000                0.576923   
Information/Explanation                     0.023077                0.007692   
Instruction                                 0.000000                0.026316   
Interview                                   0.000000                0.125000   
Invitation                                  0.000000

In [8]:
pd.crosstab(ginco["primary_level_1"], ginco["CORE_main_pred_on_MT"], normalize="index").to_dict("index")

{'Announcement': {'How-To/Instructional': 0.0,
  'Informational Description/Explanation': 0.9411764705882353,
  'Informational Persuasion': 0.0,
  'Interactive Discussion': 0.0,
  'Lyrical': 0.0,
  'Narrative': 0.058823529411764705,
  'Opinion': 0.0,
  'Spoken': 0.0},
 'Call': {'How-To/Instructional': 0.0,
  'Informational Description/Explanation': 1.0,
  'Informational Persuasion': 0.0,
  'Interactive Discussion': 0.0,
  'Lyrical': 0.0,
  'Narrative': 0.0,
  'Opinion': 0.0,
  'Spoken': 0.0},
 'Correspondence': {'How-To/Instructional': 0.0,
  'Informational Description/Explanation': 0.3125,
  'Informational Persuasion': 0.0625,
  'Interactive Discussion': 0.3125,
  'Lyrical': 0.0625,
  'Narrative': 0.0625,
  'Opinion': 0.0625,
  'Spoken': 0.125},
 'FAQ': {'How-To/Instructional': 0.0,
  'Informational Description/Explanation': 0.6666666666666666,
  'Informational Persuasion': 0.0,
  'Interactive Discussion': 0.0,
  'Lyrical': 0.0,
  'Narrative': 0.0,
  'Opinion': 0.0,
  'Spoken': 0.3333

If there is no information regarding the MT-GINCO results, they are the same as the GINCO.

1. Well connected labels:
* FAQ: 'Informational Description/Explanation': 0.67
* Forum: 'Interactive Discussion': 0.76; much worse with MT-GINCO: 'Interactive Discussion': 0.58, 'Narrative': 0.15
* 'Information/Explanation': 'Informational Description/Explanation': 0.907
* Interview: 'Spoken': 0.875
* 'Script/Drama': 'Spoken': 1.0; much worse with MT-GINCO: 'Informational Description/Explanation': 1.0
* 'Legal/Regulation': 'Informational Description/Explanation': 1.0
* Lyrical': 'Lyrical': 0.75
* 'News/Reporting': between 'Informational Description/Explanation': 0.496 and 'Narrative': 0.452; better with MT-GINCO: 'Narrative': 0.59, 'Informational Description/Explanation': 0.37
* 'Opinionated News': between 'Narrative': 0.52 and Informational Description/Explanation': 0.348; better results with MT-GINCO: 'Narrative': 0.65
* 'Prose': 'Narrative': 0.5, even better with MT-GINCO: 'Narrative': 0.67
* 'Recipe': 'How-To/Instructional': 1.0
* 'Research Article': 'Informational Description/Explanation': 1.0

2. Not well connected: 
* Announcement: 'Informational Description/Explanation': 0.94
* Call: 'Informational Description/Explanation': 1.0
* 'Correspondence': 'Interactive Discussion': 0.5; with MT-GINCO between 'Informational Description/Explanation': 0.31 and 'Interactive Discussion': 0.31
* 'Instruction': split between 'How-To/Instructional': 0.42 and 'Informational Description/Explanation': 0.53; with MT-GINCO a bit more 'How-To/Instructional': 0.5
* 'Invitation': 'Informational Description/Explanation': 0.93; even more Informational Description/Explanation with MT-GINCO:  'Informational Description/Explanation': 0.97
* 'List of Summaries/Excerpts': various categories; with MT-GINCO mostly  'Informational Description/Explanation': 0.42, 'Narrative': 0.30
* 'Opinion/Argumentation': 'Informational Description/Explanation': 0.438, 'Spoken': 0.20; even more scattered across categories with MT-GINCO: 'Informational Description/Explanation': 0.33l, 'Informational Description/Explanation': 0.33, 'Spoken': 0.22
* Other: mostly 'Informational Description/Explanation': 0.56; similar with MT-GINCO: 'Informational Description/Explanation': 0.47
* 'Promotion': 'Informational Description/Explanation': 0.7
* 'Promotion of Services':  'Informational Description/Explanation': 0.84
* 'Promotion of a Product': 'Informational Description/Explanation': 0.49; a bit "better" with MT-GINCO: 'Informational Description/Explanation': 0.51, but also 'Informational Persuasion': 0.4
*  'Review': 'Informational Description/Explanation': 0.294, various other categories; even worse with MT-GINCO: 'Spoken': 0.41



## Compare CORE-main labels with FTD

In [2]:
# Import the FTD dataset
ftd = pd.read_csv("data-sheets-with-all-info/FTD-dataset-with-all-information.csv", sep="\t", index_col=0)
ftd.head(3)

ID              labels Multiple labels  \
0  __id__1-syndicate  A1 (argumentative)             NaN   
1  __id__2-syndicate  A1 (argumentative)             NaN   
2  __id__3-syndicate  A1 (argumentative)             NaN   

                                                text  length  \
0  BMW's and Chinese Justice * * * * * In most pl...     975   
1  China and a New Balance of Power SHANGHAI – Th...     956   
2  China and Russia in the New World Disorder Can...     978   

       GINCO_downcast_pred MT-GINCO_downcast_pred CORE_main_pred  
0           News/Reporting         News/Reporting      Narrative  
1  Information/Explanation  Opinion/Argumentation        Opinion  
2    Opinion/Argumentation  Opinion/Argumentation        Opinion

In [5]:
ftd.columns

Index(['ID', 'labels', 'Multiple labels', 'text', 'length',
       'GINCO_downcast_pred', 'MT-GINCO_downcast_pred', 'CORE_main_pred'],
      dtype='object')

In [3]:
ftd["Multiple labels"].value_counts()

y    139
Name: Multiple labels, dtype: int64

In [4]:
# Discard texts with multiple labels
ftd = ftd[ftd["Multiple labels"] != "y"]
ftd.describe(include="all")

ID              labels Multiple labels  \
count                1414                1414               0   
unique               1414                  10               0   
top     __id__1-syndicate  A1 (argumentative)             NaN   
freq                    1                 296             NaN   
mean                  NaN                 NaN             NaN   
std                   NaN                 NaN             NaN   
min                   NaN                 NaN             NaN   
25%                   NaN                 NaN             NaN   
50%                   NaN                 NaN             NaN   
75%                   NaN                 NaN             NaN   
max                   NaN                 NaN             NaN   

                                                     text         length  \
count                                                1414    1414.000000   
unique                                               1414            NaN   
top     BMW's and Chinese Justice * * * * * In most pl...            NaN   
freq                                                    1            NaN   
mean                                                  NaN    1445.212164   
std                                                   NaN    4989.570842   
min                                                   NaN      31.000000   
25%                                                   NaN     224.000000   
50%                                                   NaN     495.000000   
75%                                                   NaN    1144.250000   
max                                                   NaN  146922.000000   

            GINCO_downcast_pred   MT-GINCO_downcast_pred  \
count                      1414                     1414   
unique                        9                        9   
top     Information/Explanation  Information/Explanation   
freq                        407                      311   
mean                        NaN                      NaN   
std                         NaN                      NaN   
min                         NaN                      NaN   
25%                         NaN                      NaN   
50%                         NaN                      NaN   
75%                         NaN                      NaN   
max                         NaN                      NaN   

                               CORE_main_pred  
count                                    1414  
unique                                      8  
top     Informational Description/Explanation  
freq                                      772  
mean                                      NaN  
std                                       NaN  
min                                       NaN  
25%                                       NaN  
50%                                       NaN  
75%                                       NaN  
max                                       NaN

In [6]:
# Analyse how CORE main labels are connected with FTD labels based on the prediction
pd.crosstab(ftd['labels'], ftd["CORE_main_pred"], normalize="index")


CORE_main_pred      How-To/Instructional  \
labels                                     
A1 (argumentative)              0.003378   
A11 (personal)                  0.000000   
A12 (promotion)                 0.011719   
A14 (academic)                  0.000000   
A16 (information)               0.000000   
A17 (review)                    0.000000   
A4 (fiction)                    0.000000   
A7 (instruction)                0.442424   
A8 (news)                       0.000000   
A9 (legal)                      0.000000   

CORE_main_pred      Informational Description/Explanation  \
labels                                                      
A1 (argumentative)                               0.516892   
A11 (personal)                                   0.202532   
A12 (promotion)                                  0.691406   
A14 (academic)                                   0.911392   
A16 (information)                                0.910714   
A17 (review)                                     0.235294   
A4 (fiction)                                     0.095745   
A7 (instruction)                                 0.418182   
A8 (news)                                        0.257353   
A9 (legal)                                       0.986301   

CORE_main_pred      Informational Persuasion  Interactive Discussion  \
labels                                                                 
A1 (argumentative)                  0.006757                0.013514   
A11 (personal)                      0.000000                0.113924   
A12 (promotion)                     0.167969                0.011719   
A14 (academic)                      0.000000                0.000000   
A16 (information)                   0.005952                0.000000   
A17 (review)                        0.058824                0.029412   
A4 (fiction)                        0.000000                0.010638   
A7 (instruction)                    0.006061                0.018182   
A8 (news)                           0.000000                0.000000   
A9 (legal)                          0.000000                0.000000   

CORE_main_pred       Lyrical  Narrative   Opinion    Spoken  
labels                                                       
A1 (argumentative)  0.000000   0.233108  0.216216  0.010135  
A11 (personal)      0.037975   0.481013  0.113924  0.050633  
A12 (promotion)     0.003906   0.058594  0.054688  0.000000  
A14 (academic)      0.000000   0.050633  0.037975  0.000000  
A16 (information)   0.017857   0.065476  0.000000  0.000000  
A17 (review)        0.000000   0.102941  0.558824  0.014706  
A4 (fiction)        0.021277   0.797872  0.074468  0.000000  
A7 (instruction)    0.000000   0.018182  0.096970  0.000000  
A8 (news)           0.000000   0.742647  0.000000  0.000000  
A9 (legal)          0.000000   0.013699  0.000000  0.000000

In [7]:
pd.crosstab(ftd['labels'], ftd["CORE_main_pred"], normalize="index").to_dict("index")

{'A1 (argumentative)': {'How-To/Instructional': 0.0033783783783783786,
  'Informational Description/Explanation': 0.5168918918918919,
  'Informational Persuasion': 0.006756756756756757,
  'Interactive Discussion': 0.013513513513513514,
  'Lyrical': 0.0,
  'Narrative': 0.23310810810810811,
  'Opinion': 0.21621621621621623,
  'Spoken': 0.010135135135135136},
 'A11 (personal)': {'How-To/Instructional': 0.0,
  'Informational Description/Explanation': 0.20253164556962025,
  'Informational Persuasion': 0.0,
  'Interactive Discussion': 0.11392405063291139,
  'Lyrical': 0.0379746835443038,
  'Narrative': 0.4810126582278481,
  'Opinion': 0.11392405063291139,
  'Spoken': 0.05063291139240506},
 'A12 (promotion)': {'How-To/Instructional': 0.01171875,
  'Informational Description/Explanation': 0.69140625,
  'Informational Persuasion': 0.16796875,
  'Interactive Discussion': 0.01171875,
  'Lyrical': 0.00390625,
  'Narrative': 0.05859375,
  'Opinion': 0.0546875,
  'Spoken': 0.0},
 'A14 (academic)': {

In general, FTD categories are identified well with the CORE-main classifier - 7 categories match, while 3 categories are not well predicted. As in the case of the CORE-main predictions to the GINCO dataset, we can see that the category "Opinion" is not matched with the FTD category 'A1 (argumentative)' or 'A11 (personal)' which were expected to be connected. However, here, FTD's Review is better identified as CORE's "Opinion" than in the case of GINCO. As with GINCO, the CORE-main classifier is not capable of recognizing promotion.

1. Categories that are connected well:
* 'A11 (personal)': 'Narrative': 0.48
* A14 (academic): Informational Description/Explanation': 0.91
* 'A16 (information)': 'Informational Description/Explanation': 0.91
* 'A17 (review)': 'Opinion': 0.56
* 'A4 (fiction)': 'Narrative': 0.80
* 'A8 (news)': 'Narrative': 0.74
* 'A9 (legal)': 'Informational Description/Explanation': 0.99

2. Categories that are not connected well/wrongly predicted:
* 'A1 (argumentative)': 'Informational Description/Explanation': 0.51,  'Opinion': 0.22
* 'A12 (promotion)': 'Informational Description/Explanation': 0.69
* 'A7 (instruction)': 'How-To/Instructional': 0.44, 'Informational Description/Explanation': 0.42